packages/jest-haste-map/src/lib/extractRequires.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+const blockCommentRe = /\/\*[^]*?\*\//g;
+const lineCommentRe = /\/\/.*/g;
+
+/* eslint-disable max-len */
+const replacePatterns = {
+  EXPORT_RE: /(\bexport\s+(?:[^'"]+\s+from\s+)??)(['"])([^'"]+)(\2)/g,
+  IMPORT_RE: /(\bimport\s+(?:[^'"]+\s+from\s+)??)(['"])([^'"]+)(\2)/g,
```

Comment:
For instance, if you have a string like this:

```js
import foo from 'foo"bar';
```

the regex here won't match. If you use this instead:

```js
/(\bimport\s+(?:[^'"]+\s+from\s+)??)(['"])(.+?)(\2)/g
```

it will match. As I said, in practice it is probably fine, but this seems better.

```diff
- /(\bimport\s+(?:[^'"]+\s+from\s+)??)(['"])([^'"]+?)(\2)/g
+ /(\bimport\s+(?:[^'"]+\s+from\s+)??)(['"])(.+?)(\2)/g